tidy3d/plugins/adjoint/components/geometry.py

```diff
@@ -237,7 +237,7 @@ def store_vjp(
                     eps_data = grad_data_eps.field_components[eps_field_name].isel(f=0)
 
                     # get the permittivity values just inside and outside the edge
-                    n_cells_in = 3
+                    n_cells_in = 2
```

Comment:
maybe a naive question: why is it 2 not 1?

Reply:
it's not obvious, actually. 0 corresponds to just outside of the box (since the permittivity monitor extends a bit beyond the bounds), 1 corresponds usually to the subpixelled grid cell. we want to grab the grid cell that is completely inside of the object, which is at least 2 (at least it was). When the box is very thin, it has 3 grid cells, so it becomes complicated. 

Maybe I can ask you, if you sub pixel a very thin box (smaller than a grid cell) does the structure only influence one grid cell on the yee lattice?

Reply:
I think it can influence 0 or 1 or 2:

- 0: if it's so thin that our sampling doesn't catch it
- 1: it's completely within a cell
- 2: it's sandwiched between 2 cells.